presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -227,6 +227,10 @@ protected Node visitShowCatalogs(ShowCatalogs node, Void context)
                     .map(name -> row(new StringLiteral(name)))
                     .collect(toList());
 
+            if (rows.isEmpty()) {
```

Comment:
It's needed so as not to return NULL when there are no catalogs to be shown. When rows is an empty list, the FROM relation becomes empty in the SELECT query and the query fails, returning null.